src/main/java/org/cyclops/cyclopscore/tileentity/CyclopsTileEntity.java

```diff
@@ -293,23 +287,25 @@ protected EnumFacing transformFacingForRotation(EnumFacing facing) {
 
     @Override
     public boolean hasCapability(@Nonnull Capability<?> capability, EnumFacing facing) {
-        return (sidedCapabilities != null && sidedCapabilities.contains(capability, transformFacingForRotation(facing)))
-                || (capabilities != null && capabilities.containsKey(capability))
-                || super.hasCapability(capability, facing);
+        if (facing != null){
```

Comment:
You can inline all of this.

Reply:
how?

Reply:
Sorry, but I don't have the time to hold your hand like this, I'm not here to help you code.

Reply:
return facing != null ? sidedCapabilities.contains(capability, facing) : innerCapabilities.containsKey(capability) || super.hasCapability(capability, facing);